File: Java Programs/SortingAlgos/InsertionSort.java
import java.util.*;

class InsertionSort {

    public static int[] insertionSort(int[] arr) {
        for(int i = 1; i<arr.length; i++) {
            int current = arr[i];
            int j = i-1;
            while(j>=0 && current < arr[j]) {
                arr[j+1] = arr[j];
                j--;
            }

            arr[j+1] = current;
        }

        return arr;
    }

    public static void main(String[] args) {
        int[] arr = {7,8,3,1,2,-1,-4,0};
        int[] sorted_arr = insertionSort(arr);

        for(int i = 0; i < sorted_arr.length; i++) {
            System.out.print(sorted_arr[i] + " ");
        }
    }
}

File: Python/Sorting Algos/InsertionSort.py
def insertionSort(arr):
    for i in range(1,len(arr)):
        current = arr[i]
        j = i-1

        while(j>=0 and current < arr[j]):
            arr[j+1] = arr[j]
            j -= 1

        # Placement
        arr[j+1] = current

    return arr

if __name__ == '__main__':
    arr = [7,8,1,3,2,-1,-4,-2,0]
    print(insertionSort(arr))
